detection-rules/headers_voicemail_sendgrid.yml

```diff
@@ -5,8 +5,12 @@ reference:
   - "https://krebsonsecurity.com/2020/08/sendgrid-under-siege-from-hacked-accounts/comment-page-1/"
 type: "rule"
 source: |
-  type.inbound
-  and headers.return_path.domain.domain == 'sendgrid.net'
-  and ilike(subject.subject, '*voicemail*', '*voice message*')
+  type.inbound 
+  and (headers.return_path.domain.domain == 'sendgrid.net'
+    or any(headers.domains, ilike(.domain, "*.outbound-mail.sendgrid.net"))
+    or any(headers.hops, any(.fields, ilike(.name, "Received") and ilike(.value, "*.outbound-mail.sendgrid.net*")))
+    or any(headers.hops, any(.authentication_results.dkim_details, .domain == 'sendgrid.net'))
```

Comment:
@jkamdjou are these redundant with the `headers.domains` check? I believe that both these locations are copied into `headers.domains`, which is also much faster. If `headers.domains` isn't too permissive, then we would only need to do `any(headers.domains, ilike(.domain, "*.outbound-mail.sendgrid.net"))`.

Reply:
`headers.domains` aggregates the domains found in the [`Received` headers only](https://docs.sublimesecurity.com/docs/mdm), which wouldn't include authentication results. but i do think `headers.domains` should probably be sufficient on its own, unless @arnydo is seeing some instances where one is present and not the other

<img width="584" alt="image" src="https://user-images.githubusercontent.com/11003450/178019664-c5d2c14a-8137-486e-b0e6-8ef96ff2a1df.png">

Reply:
I am pretty sure it wasn't firing until I added the additional check. Let me verify again and let you know.